bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -68,6 +73,18 @@ impl Client {
         })
     }
 
+    /// Checks if a store path has been set and throws an error if not.
+    fn ensure_store_path(&self) -> anyhow::Result<()> {
+        self.store_path()
+            .map(|_| ())
+            .ok_or_else(|| anyhow!("Login failed: Make sure to set a store path on the client."))
+    }
```

Comment:
I'd much rather expect this to be something the builder would enforce ... like... if you want this the `ClientBuilder::build()` should already fail, not the login or restore fn...